lua/gitlab/utils/init.lua

```diff
@@ -485,10 +485,10 @@ end
 ---@param settings table|nil User defined popup settings
 ---@param width number? Override default width
 ---@param height number? Override default height
+---@param zindex number? Override default zindex
 ---@return table
 M.create_popup_state = function(title, settings, width, height, zindex)
-  local default_settings = require("gitlab.state").settings.popup
-  local user_settings = settings or {}
+  settings = M.merge(require("gitlab.state").settings.popup, settings or {})
```

Comment:
Please do not variable shadow, either call the parameter something else or call this `merged_settings`